### PR TITLE
[WIP] Followup to fix RHOAIENG-556 correctly - Properly dry-run all operations in KServe modal before submitting

### PR DESCRIPTION
**Pull request by mturley**
Resolves https://issues.redhat.com/browse/RHOAIENG-556

@lucferbux realized the changes in https://github.com/opendatahub-io/odh-dashboard/pull/2319 didn't actually fix this issue, we both had the logic backwards when reasoning about it and testing it.

WIP -- needs some further refactoring and new tests.

## Request review criteria:

Self checklist (all need to be checked):
- [ ] The developer has manually tested the changes and verified that the changes work
- [ ] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx [hunk 1]
```diff
@@ -2,7 +2,8 @@ import * as React from 'react';
 import { Form, FormSection, Modal, Stack, StackItem } from '@patternfly/react-core';
 import { EitherOrNone } from '@openshift/dynamic-plugin-sdk';
 import {
-  submitInferenceServiceResource,
+  getSubmitInferenceServiceResourceFn,
```
Comment (lucferbux) [inline frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:5]: Fix this linting issue

File: frontend/src/pages/modelServing/screens/projects/kServeModal/ManageKServeModal.tsx [hunk 2]
```diff
@@ -159,16 +159,27 @@ const ManageKServeModal: React.FC<ManageKServeModalProps> = ({
...
+    const submitInferenceServiceResource = getSubmitInferenceServiceResourceFn(
+      createDataInferenceService,
+      editInfo?.inferenceServiceEditInfo,
+      servingRuntimeName,
+      false,
+      acceleratorProfileState,
+      replicaCount,
+    );
+
+    // TODO mturley add unit test coverage to make sure we don't create any resources if there's an error
```
Comment (lucferbux) [inline frontend/src/pages/modelServing/screens/projects/kServeModal/ManageKServeModal.tsx:173]: 👍

File: frontend/src/pages/modelServing/screens/projects/utils.ts [hunk 3]
```diff
@@ -288,24 +291,27 @@ const createInferenceServiceAndDataConnection = (
...
         undefined,
         isModelMesh,
         acceleratorProfileState,
         replicaCount,
+        dryRun,
       );
 };
 
-export const submitInferenceServiceResource = (
+// TODO mturley we need a dryRun here too right?
```
Comment (lucferbux) [inline frontend/src/pages/modelServing/screens/projects/utils.ts:306]: I guess you forgot to remove this TODO right? I think returning the function with the dry run is a great approach here. Are you considering something else?

File: frontend/src/pages/modelServing/screens/projects/utils.ts [hunk 4]
```diff
@@ -358,73 +376,81 @@ export const submitServingRuntimeResources = async (
...
-          ),
-        ]
-      : [
-          createServingRuntime({
-            data: servingRuntimeData,
-            namespace,
-            servingRuntime: servingRuntimeSelected,
-            isCustomServingRuntimesEnabled: customServingRuntimesEnabled,
-            opts: {
+  // TODO mturley Lucas needed to update this condition, there'll be a merge conflict here
```
Comment (lucferbux) [inline frontend/src/pages/modelServing/screens/projects/utils.ts:379]: Yeah don't worry about that, we can work through this

File: frontend/src/pages/modelServing/screens/projects/utils.ts (hunk 3 above)
Comment (mturley) [reply frontend/src/pages/modelServing/screens/projects/utils.ts:306]: Yeah I just forgot to remove this

**Comment by dpanshug** [comment]
Closing this PR, creating a new one with changes added.

**Pull request closed without merging**